app/uk/gov/hmrc/economiccrimelevyregistration/controllers/EntityTypeController.scala

```diff
@@ -123,17 +127,20 @@ class EntityTypeController @Inject() (
       url => Redirect(Call(GET, url))
     )
 
-  private def cleanup(
+  private def cleanupIfChange(
     registration: Registration,
-    newEntityType: EntityType
+    newEntityType: EntityType,
+    previousEntityType: Option[EntityType]
   ) =
-    if (EntityType.isOther(newEntityType)) {
-      dataCleanup.cleanupOtherEntityData(
-        registration.copy(entityType = Some(newEntityType))
-      )
+    if (previousEntityType.contains(newEntityType)) {
```

Comment:
This would be easier to read in the functional style. See line 104

Reply:
Changed.